src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -168,7 +168,12 @@ internal void ProcessLogsFromHttpResponse(ScriptInvocationContext scriptInvocati
 
         private void AddRequestHeadersAndSetRequestUri(HttpRequestMessage httpRequestMessage, string functionName, string invocationId)
         {
-            httpRequestMessage.RequestUri = new Uri(new UriBuilder(WorkerConstants.HttpScheme, WorkerConstants.HostName, _httpWorkerOptions.Port, functionName).ToString());
+            string pathValue = functionName;
+            if (httpRequestMessage.RequestUri != null)
+            {
+                pathValue = httpRequestMessage.RequestUri.AbsolutePath.Replace("api", string.Empty);
```

Comment:
Will this work if someone overrides the `routePrefix` in host.json?

Reply:
Also, `api` can appear elsewhere in the path and in multiple places.

Reply:
Ideally, we should forward the request to the same route as in the original request. And yes, this will work if `routePrefix` is provided in host.json